ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -135,7 +145,7 @@ private List<Partition> getPartitions(Table table, AlterTableCompactDesc desc, D
     List<Partition> partitions = new ArrayList<>();
 
     if (desc.getPartitionSpec() == null) {
-      if (table.isPartitioned()) {
+      if (table.isPartitioned() || (DDLUtils.isIcebergTable(table) && table.getStorageHandler().isPartitioned(table))) {
```

Comment:
why check for iceberg?

Reply:
because `table.isPartitioned()` does not work for Iceberg. For Iceberg it will always return False. That's why I added  a call to getStorageHandler().isPartitioned(table).